src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,311 @@
+	 *
+	 * @since 6.3.0
+	 */
+	public function __construct() {
+		$this->parent_post_type = 'wp_global_styles';
+		$this->rest_base        = 'revisions';
+		$this->parent_base      = 'global-styles';
+		$this->namespace        = 'wp/v2';
+	}
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent of the revision.' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Returns revisions of the given global styles config custom post type.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param WP_REST_Request $request The request instance.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_items( $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+
+		if ( is_wp_error( $parent ) ) {
+			return $parent;
+		}
+		$response                         = array();
+		$raw_config                       = json_decode( $parent->post_content, true );
+		$is_global_styles_user_theme_json = isset( $raw_config['isGlobalStylesUserThemeJSON'] ) && true === $raw_config['isGlobalStylesUserThemeJSON'];
+
+		if ( $is_global_styles_user_theme_json ) {
+			$user_theme_revisions = wp_get_post_revisions(
+				$parent->ID,
+				array(
+					'posts_per_page' => 100,
+				)
+			);
+
+			if ( ! empty( $user_theme_revisions ) ) {
+				foreach ( $user_theme_revisions as $revision ) {
+					$revision   = $this->prepare_item_for_response( $revision, $request );
+					$response[] = $this->prepare_response_for_collection( $revision );
+				}
+			}
+		}
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Prepares the revision for the REST response.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param WP_Post         $item    Post revision object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+		// Retrieves global styles config as JSON.
+		$raw_revision_config = json_decode( $item->post_content, true );
+		$config              = ( new WP_Theme_JSON( $raw_revision_config, 'custom' ) )->get_raw_data();
+
+		// Prepares item data.
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( rest_is_field_included( 'author', $fields ) ) {
+			$data['author'] = (int) $item->post_author;
+		}
+
+		if ( rest_is_field_included( 'date', $fields ) ) {
+			$data['date'] = $item->post_date;
```

Comment:
Just checking, should I also copy across the date formatting functions?

```php
		if ( in_array( 'date', $fields, true ) ) {
			$data['date'] = $this->prepare_date_response( $post->post_date_gmt, $post->post_date );
		}
```

See: https://github.com/WordPress/wordpress-develop/blob/82b6dddc066c5b5e3219eecc60c3051317de3d24/src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L564-L564

Reply:
Assuming "yes" 😄